unittests/Dialect/ESI/runtime/ESIRuntimeTest.cpp

```diff
@@ -0,0 +1,449 @@
+//===- ESIRuntimeTest.cpp - ESI Runtime Type System Tests ---------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "esi/Types.h"
+#include "gtest/gtest.h"
+#include <any>
+#include <cstdint>
+#include <map>
+#include <vector>
+
+using namespace esi;
+
+namespace {
+
+// Test VoidType serialization and deserialization
+TEST(ESITypesTest, VoidTypeSerialization) {
+  VoidType voidType("void");
+
+  // Test valid void value (empty std::any)
+  std::any voidValue;
+  EXPECT_NO_THROW(voidType.isValid(voidValue));

```

Comment:
`isValid` doesn't ever throw so `EXPECT_NO_THROW` will always evaluate true. This issue is repeated a bunch of times in this test.